File: app/src/main/java/com/example/lovelink/SignUpActivity.java
package com.example.lovelink;

import android.app.Activity;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class SignUpActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sign_up);

        // Referencias a los campos del layout
        EditText phoneEditText = findViewById(R.id.phoneEditText); // Este campo es el teléfono
        EditText emailEditText = findViewById(R.id.emailEditText);
        EditText passwordEditText = findViewById(R.id.passwordEditText);
        EditText password2EditText = findViewById(R.id.password2EditText);
        Button signupButton = findViewById(R.id.signupButton);

        signupButton.setOnClickListener(v -> {
            String phone = phoneEditText.getText().toString().trim();
            String email = emailEditText.getText().toString().trim();
            String password = passwordEditText.getText().toString();
            String password2 = password2EditText.getText().toString();

            if (phone.isEmpty() || email.isEmpty() || password.isEmpty() || password2.isEmpty()) {
                Toast.makeText(SignUpActivity.this, "Rellena todos los campos", Toast.LENGTH_SHORT).show();
            } else if (!password.equals(password2)) {
                Toast.makeText(SignUpActivity.this, "Las contraseñas no coinciden", Toast.LENGTH_SHORT).show();
            } else {
                // Guardar los datos temporalmente en SharedPreferences para ser utilizados después
                SharedPreferences.Editor editor = getSharedPreferences("LoveLinkPrefs", MODE_PRIVATE).edit();
                editor.putBoolean("isLoggedIn", true);
                editor.putString("phone", phone);
                editor.putString("email", email);
                editor.putString("password", password);
                editor.apply();

                // Pasar los datos a la primera pantalla (Formulario 1)
                Intent intent = new Intent(SignUpActivity.this, ProfileSetup1Activity.class);
                intent.putExtra("phone", phone);
                intent.putExtra("email", email);
                intent.putExtra("password", password);
                startActivity(intent);
                finish();
            }
        });

    }
}

File: app/src/main/java/com/example/lovelink/SignUpActivity.kt
// SignUpActivity.kt
package com.example.lovelink

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import com.example.lovelink.models.Cuenta
import com.example.lovelink.network.RetrofitClient
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.security.MessageDigest

class SignUpActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sign_up)

        val phoneEditText = findViewById<EditText>(R.id.phoneEditText)
        val emailEditText = findViewById<EditText>(R.id.emailEditText)
        val passwordEditText = findViewById<EditText>(R.id.passwordEditText)
        val password2EditText = findViewById<EditText>(R.id.password2EditText)
        val signupButton = findViewById<Button>(R.id.signupButton)

        signupButton.setOnClickListener {
            val telefono = phoneEditText.text.toString().trim()
            val email = emailEditText.text.toString().trim()
            val password = passwordEditText.text.toString()
            val password2 = password2EditText.text.toString()

            if (telefono.isEmpty() || email.isEmpty() || password.isEmpty() || password2.isEmpty()) {
                Toast.makeText(this, "Rellena todos los campos", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            if (!Regex("^6\\d{8}$").matches(telefono)) {
                Toast.makeText(this, "El teléfono debe comenzar por 6 y tener 9 dígitos", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            if (!android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
                Toast.makeText(this, "Correo electrónico no válido", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            if (password.length < 6 || !password.contains(Regex("[^A-Za-z0-9]"))) {
                Toast.makeText(this, "La contraseña debe tener al menos 6 caracteres y un carácter especial", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            if (password != password2) {
                Toast.makeText(this, "Las contraseñas no coinciden", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            val hashedPassword = hashSHA256(password)

            val cuenta = Cuenta(
                telefono = telefono,
                email = email,
                password = hashedPassword
            )

            RetrofitClient.cuentaService.registrarCuenta(cuenta).enqueue(object : Callback<Cuenta> {
                override fun onResponse(call: Call<Cuenta>, response: Response<Cuenta>) {
                    if (response.isSuccessful) {
                        val cuentaCreada = response.body()
                        if (cuentaCreada != null) {
                            val intent = Intent(this@SignUpActivity, ProfileSetup1Activity::class.java)
                            intent.putExtra("cuenta_id", cuentaCreada.id)
                            startActivity(intent)
                            finish()
                        }
                    } else {
                        val errorBody = response.errorBody()?.string()
                        when {
                            errorBody?.contains("telefono") == true -> {
                                Toast.makeText(this@SignUpActivity, "El teléfono ya está registrado", Toast.LENGTH_SHORT).show()
                            }
                            errorBody?.contains("email") == true -> {
                                Toast.makeText(this@SignUpActivity, "El correo ya está registrado", Toast.LENGTH_SHORT).show()
                            }
                            else -> {
                                Toast.makeText(this@SignUpActivity, "Error al registrar cuenta", Toast.LENGTH_SHORT).show()
                            }
                        }
                        Log.e("SignUp", "Error: ${response.code()} - $errorBody")
                    }
                }

                override fun onFailure(call: Call<Cuenta>, t: Throwable) {
                    Toast.makeText(this@SignUpActivity, "Fallo de red: ${t.message}", Toast.LENGTH_LONG).show()
                    Log.e("SignUp", "Error de red", t)
                }
            })
        }
    }

    private fun hashSHA256(input: String): String {
        val bytes = MessageDigest.getInstance("SHA-256").digest(input.toByteArray())
        return bytes.joinToString("") { "%02x".format(it) }
    }
}
